packages/lit-dev-content/site/tutorials/content/word-viewer/01.md

```diff
@@ -0,0 +1,74 @@
+
+
+Define a Lit component by creating a class extending `LitElement` and
+registering your class with the browser:
+
+{% switchable-sample %}
+
+```ts
+@customElement('word-viewer')
+export class WordViewer extends LitElement { /* ... */ }
+```
+
+```js
+export class WordViewer extends LitElement { /* ... */  }
+customElements.define('word-viewer', WordViewer);
+```
+
+{% endswitchable-sample %}
+
+The TypeScript `@customElement` decorator is a shorthand for the JavaScript
+equivalent of calling `customElements.define`, which registers the custom
+element class with the browser and associates it with the custom element name
+`word-viewer`.
+
+<litdev-aside type="positive">
+
+A [valid custom element
+name](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name)
+must contain a hyphen.
+
+This ensures forward compatibility with future tags added to HTML or SVG which
+won't add hyphen containing elements.
+
+</litdev-aside>
+
+## Rendering
+
+A Lit component's `render()` method returns a
+[template](/docs/templates/overview/). Lit templates use the
+`html` [tagged template
+literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#tagged_templates)
+to run without a build step in JavaScript.
+
+Add a template to define what `word-viewer` should render:
+
+```ts
+import { html, LitElement } from 'lit';
+...
+
+  render() {
+    return html`<pre>A super expressive and efficient template!</pre>`
+  }
+```
+
+## Optional: Providing good TypeScript types
```

Comment:
Up to you if you want to do this but this component by @e111077 just got merged which allows dynamic content outside of code samples based on language selection so you could hide this for JS viewers.
https://github.com/lit/lit.dev/blob/2cbcb3486c7b59f5a2ec6489b321c5ba9b890753/packages/lit-dev-content/src/components/ts-js.ts#L24

Reply:
Great point! This section is small enough that maybe it's interesting for JS readers, but could also be distracting.

Elliott - what's most consistent or preferred?

Reply:
If you prefer to keep it for JS readers I say use a details/summary

```html
<details>
<summary><h2>Optional:...</h2></summary>

markdown markdowm markdown

</details>`

Reply:
The details summary isn't inline with the arrow. Opting to use the new `ts-js`.

![Screen Shot 2022-04-28 at 10 18 04 AM](https://user-images.githubusercontent.com/15080861/165810241-386813f2-97c7-43d5-9569-bcd5bb247d02.png)

Reply:
Weird, what was the markup for that? ## or h2

Reply:
This was using:

```
<details>
<summary><h2>Optional:...</h2></summary>
...
```

ts-js is working great.